app/workers/facilities/pssg_download.rb

```diff
@@ -56,19 +56,31 @@ def extract_polygon(rings)
     end
 
     def insert_or_update_band(facility)
-      if facility.drivetime_bands.exists?(vha_facility_id: @id, name: @band_name)
-        Rails.logger.info "PSSG Band not updated: Facility #{@id}. Band #{@attributes&.dig('Name')}"
-      else
+      drive_time_band = facility.drivetime_bands.find_by(vha_facility_id: @id, name: @band_name)
+
+      if drive_time_band.nil?
         drive_time_band = facility.drivetime_bands.new(vha_facility_id: @id, name: @band_name)
         drive_time_band.unit = 'minutes'
         drive_time_band.min = round_band(@attributes&.dig('FromBreak'))
         drive_time_band.max = round_band(@attributes&.dig('ToBreak'))
-        drive_time_band.name = @band_name
         drive_time_band.polygon = extract_polygon(@rings)
+        drive_time_band.vssc_extract_date = @vssc_extract_date
 
-        Rails.logger.info "PSSG Band successfully saved: #{@band_name}" # temporary logging
         drive_time_band.save
```

Comment:
could this be?
```
        facility.drivetime_bands.create(
          vha_facility_id: @id,
          name: @band_name,
          unit: 'minutes',
          min: round_band(@attributes&.dig('FromBreak')),
          max: round_band(@attributes&.dig('ToBreak')),
          polygon: extract_polygon(@rings),
          vssc_extract_date: @vssc_extract_date
        )
```

Reply:
`create` could definitely replace setting the properties individually for a new record and then saving (lines 62-69). I'm not clear on what `persisted?` is doing though. Won't it always return `false`?